### PR TITLE
Unit test and example for user to connect to cloud db

### LINKED ISSUE #436: Open a discussion on GitHub Actions / Server Status for Cloud MongoDB
Our documentation has a [how-to](https://watertap.readthedocs.io/en/latest/how_to_guides/how_to_use_EDB.html) section on ways to connect to and utilize the Cloud version of our electrolyte database. At the most recent Electro-chem sub-team meeting we discussed whether or not we want to...

- Test connections to this cloud database through GitHub Actions 
- Provide some form of 'server status' info for the front page of WaterTap 

Are these things that we want to consider? What are the issues in trying to implement something like this?

### PR DESCRIPTION
**Pull request by aladshaw3**
## Fixes/Addresses:

#436 

## Summary/Motivation:

Creates a way to unit test connection to cloud database and uses that database to create a correctly formatted IDAES config dictionary. 

## Changes proposed in this PR:
- Added example file showing how to connect to cloud db
- Added unit test to verify that the code does connect to the database and the info in the database is correct
- Added tests for expected failure behavior

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/examples/edb/connect_to_cloud_db.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+DB_NAME = "electrolytedb"
+
+BAD_USER_NAME = "nawiedb"
+BAD_DB_NAME = "edb"
+
+public_cloud_url = f"mongodb+srv://{USER_NAME}:{PASSWORD}@nawi-edb.utpac.mongodb.net"
+
+bad_cloud_url = f"mongodb+srv://{BAD_USER_NAME}:{PASSWORD}@nawi-edb.utpac.mongodb.net"
+
+def connect_to_cloud_edb(test_invalid_host=False):
```
Comment (dangunter) [inline watertap/examples/edb/connect_to_cloud_db.py]: the name 'test_invalid_host' is a bit misleading. Really it's testing that there is a functioning MongoDB server at that host and IP that accepts connections from the given user credentials. Yes, an invalid host does violate that, but just pointing at mongodb+srv://{something}@www.mit.edu will also fail despite the fact that www.mit.edu is a fine host (according to MIT people, the finest). I would say 'check_valid_server' would be better, or to minimize the number of different ways of saying things simply 'check_connection'

**Review by dangunter — CHANGES_REQUESTED** [review]
relatively minor comment on naming.
But channeling Ludovico, I think most of the one-off functions could be pytest fixtures, probably reducing the overall code a bit

**Comment by ksbeattie** [comment]
@lbianchi-lbl suggests this might be better as a notebook.

**Comment by aladshaw3** [comment]
This is being replaced by #549

**Comment by aladshaw3** [comment]
Closed: Replaced by #549

**Pull request closed without merging**